Default values in query variables ignored

Hello, I am trying to use default values in query arguments as described here: http://graphql.org/learn/queries/#default-variables

But I am not seeing them make it through to the resolvers.

Here is a test case using an existing ```graphql_test.go``` resolver:

```
func TestHelloSnakeDefaultArguments(t *testing.T) {
        gqltesting.RunTests(t, []*gqltesting.Test{
                {
                        Schema: graphql.MustParseSchema(`
                                schema {
                                        query: Query
                                }

                                type Query {
                                        say_hello(full_name: String): String!
                                }
                        `, &helloSnakeResolver2{}),
                        Query: `
                                query($nameWithDefault: String = "Query Default", $nameNoDefault: String!) {
                                        a: say_hello(full_name: $nameWithDefault)
                                        b: say_hello(full_name: $nameNoDefault)
                                }
                        `,
                        Variables: map[string]interface{}{
                                "nameNoDefault": "Variable Supplied",
                        },

                        ExpectedResult: `
                                {
                                        "a": "Hello Query Default",
                                        "b": "Hello Variable Supplied"
                                }
                        `,
                },
        })
}
```

Currently this results in:

```
$ go test -v graphql_test.go -run TestHelloSnakeDefaultArguments
=== RUN   TestHelloSnakeDefaultArguments
--- FAIL: TestHelloSnakeDefaultArguments (0.00s)
	testing.go:40: graphql: graphql: got null for non-null
FAIL
exit status 1
FAIL	command-line-arguments	0.012s
```

If you modify SayHello to not require a full_name, you instead get whatever behavior nil results in:

```
$ go test -v graphql_test.go -run TestHelloSnakeDefaultArguments
=== RUN   TestHelloSnakeDefaultArguments
--- FAIL: TestHelloSnakeDefaultArguments (0.00s)
	testing.go:47: got:  {"a":"Hello nobody!","b":"Hello Variable Supplied!"}
	testing.go:48: want: {"a":"Hello Query Default","b":"Hello Variable Supplied"}
FAIL
exit status 1
FAIL	command-line-arguments	0.012s
```

Please let me know if there's anything I can do to help, or if I misunderstand the spec, or there's another way I'm supposed to be accessing the default value, thank you!

--- a/graphql.go
+++ b/graphql.go
@@ -154,6 +154,13 @@ func (s *Schema) exec(ctx context.Context, queryString string, operationName str
 		return &Response{Errors: []*errors.QueryError{errors.Errorf("%s", err)}}
 	}
 
+	// Fill in variables with the defaults from the operation
+	for _, v := range op.Vars {
+		if _, ok := variables[v.Name.Name]; !ok && v.Default != nil {
+			variables[v.Name.Name] = v.Default.Value(nil)
+		}
+	}
+
 	r := &exec.Request{
 		Request: selected.Request{
 			Doc:    doc,

--- a/graphql.go
+++ b/graphql.go
@@ -155,6 +155,9 @@ func (s *Schema) exec(ctx context.Context, queryString string, operationName str
 	}
 
 	// Fill in variables with the defaults from the operation
+	if variables == nil {
+		variables = make(map[string]interface{}, len(op.Vars))
+	}
 	for _, v := range op.Vars {
 		if _, ok := variables[v.Name.Name]; !ok && v.Default != nil {
 			variables[v.Name.Name] = v.Default.Value(nil)

--- a/graphql_test.go
+++ b/graphql_test.go
@@ -61,6 +61,12 @@ func (r *timeResolver) AddHour(args struct{ Time graphql.Time }) graphql.Time {
 	return graphql.Time{Time: args.Time.Add(time.Hour)}
 }
 
+type echoResolver struct{}
+
+func (r *echoResolver) Echo(args struct{ Value *string }) *string {
+	return args.Value
+}
+
 var starwarsSchema = graphql.MustParseSchema(starwars.Schema, &starwars.Resolver{})
 
 func TestHelloWorld(t *testing.T) {
@@ -435,6 +441,28 @@ func TestVariables(t *testing.T) {
 					"hero": {
 						"name": "Luke Skywalker"
 					}
+				}
+			`,
+		},
+
+		{
+			Schema: graphql.MustParseSchema(`
+				schema {
+					query: Query
+				}
+
+				type Query {
+					echo(value: String): String
+				}
+			`, &echoResolver{}),
+			Query: `
+				query Echo($value:String = "default"){
+					echo(value:$value)
+				}
+			`,
+			ExpectedResult: `
+				{
+					"echo": "default"
 				}
 			`,
 		},

